apollo-router-core/src/query_planner/mod.rs

```diff
@@ -253,66 +270,129 @@ impl PlanNode {
     }
 }
 
-/// A fetch node.
-#[derive(Debug, PartialEq, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub(crate) struct FetchNode {
-    /// The name of the service or subgraph that the fetch is querying.
-    service_name: String,
+mod fetch {
+    use std::sync::Arc;
 
-    /// The data that is required for the subgraph fetch.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    requires: Option<Vec<Selection>>,
+    use super::selection::{select_object, Selection};
+    use futures::StreamExt;
+    use serde::Deserialize;
+    use serde_json::{Map, Value};
+    use tracing::{instrument, Instrument};
 
-    /// The variables that are used for the subgraph fetch.
-    variable_usages: Vec<String>,
+    use crate::{FetchError, Object, Path, Request, Response, Schema, ServiceRegistry, ValueExt};
```

Comment:
:yarn: since the API is flat. I tend to use this everywhere so I don't have to maintain a list of imports.

```suggestion
    use crate::prelude::graphql::*;
```